error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -1166,6 +1167,23 @@ Flux<T> after(Flux<T> flux, Predicate<? super T> predicate, Comparator<? super T
     }
   }
 
+  /**
+   * Prefer {@link Flux#collect(Collector)} with {@link ImmutableList#toImmutableList} over {@link
+   * Flux#collectList()}.
+   */
+  static final class FluxCollectListToCollectImmutableList<T> {
+    @BeforeTemplate
+    Mono<List<T>> before(Flux<T> flux) {
```

Comment:
We can add: `flux.collect(toImmutableList()).map(ImmutableList::copyOf);`.

Reply:
Hmm, the `((Mono<ImmutableList<T>>) flux).map(ImmutableList::copyOf)` simplification should be in a rule of its own. (And is in fact a special case of a large range of rules.)